### PR TITLE
Add desktopitems for some classic games

### PR DESCRIPTION
**Pull request by svanderburg**
###### Motivation for this change

I'm having a difficult time to convince some of my friends that Linux/NixOS is a good OS for gaming, because these classic games cannot be conveniently started from the KDE/GNOME/XFCE and any other XDG compliant desktop :)

I have already locally created desktop items for them, but maybe it wouldn't hurt to share/upstream them.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [X] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Relase notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by wamserma — APPROVED** [review]

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 132826` at 038d054d run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built successfully:</summary>
  <ul>
    <li>dhewm3</li>
    <li>gzdoom</li>
    <li>quakespasm</li>
  </ul>
</details>
<details>
  <summary>5 suggestions:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/license-missing.md">license-missing</a>

Package is missing a license.

Near pkgs/games/quakespasm/default.nix:38:3:
```
   |
38 |   meta = with lib; {
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/games/gzdoom/default.nix:88:7:
```
   |
88 |       license = licenses.gpl3;
   |       ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall`.

Near pkgs/games/gzdoom/default.nix:67:5:
```
   |
67 |     installPhase = ''
   |     ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/patch-phase.md">patch-phase</a>

`patchPhase` should not be overridden, use `postPatch` instead.

Near pkgs/games/dhewm3/default.nix:16:3:
```
   |
16 |   patchPhase = ''
   |   ^
```
</li>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/unclear-gpl.md">unclear-gpl</a>

`gpl3` is a deprecated license, please check if project uses `gpl3Plus` or `gpl3Only` and change `meta.license` accordingly.

Near pkgs/games/dhewm3/default.nix:41:5:
```
   |
41 |     license = lib.licenses.gpl3;
   |     ^
```
</li>
  </ul>
</details>

**Review by Lassulus — APPROVED** [review]

**Comment by Lassulus** [comment]
Result of `nixpkgs-review pr 132826` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>3 packages built:</summary>
  <ul>
    <li>dhewm3</li>
    <li>gzdoom</li>
    <li>quakespasm</li>
  </ul>
</details>

File: pkgs/games/gzdoom/default.nix [hunk 1]
```diff
@@ -55,6 +55,15 @@ let
...
+      (makeDesktopItem {
+        name = "gzdoom";
+        exec = "gzdoom";
+        desktopName = "GZDoom";
+        categories = "Game;";
+      })
+    ];
+
     installPhase = ''
       install -Dm755 gzdoom "$out/lib/gzdoom/gzdoom"
```
Comment (SuperSandro2000) [inline pkgs/games/gzdoom/default.nix:68]:
```suggestion
      runHook preInstall

      install -Dm755 gzdoom "$out/lib/gzdoom/gzdoom"
```

Comment (svanderburg) [reply pkgs/games/gzdoom/default.nix:68]: Ah, I made this change locally, but forgot to commit

Comment (svanderburg) [reply pkgs/games/gzdoom/default.nix:70]: I just fixed it

**Comment by svanderburg** [comment]
If there are no further comments, I'll merge later today

**Pull request merged**